Source/UI/PaymentForm/View/ExpiryDate/ExpiryDateFormatter.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  ExpiryDateFormatter.swift
+//
+//
+//  Created by Alex Ioja-Yang on 16/05/2023.
+//
+import Foundation
+import Checkout
+
+struct ExpiryDateFormatter {
+
+    let separator: String
+    let dateFormatTextCount: Int
+    private let cardValidator: CardValidating
+
+    init(componentSeparator: String = "/",
+         cardValidator: CardValidating = CardValidator(environment: .sandbox)) {
```

Comment:
Can you remind me why we hardcod CardValidator to use .sandbox ? also would be worth adding a comment.

Reply:
We don't actually hardcode the CardValidator, this is just the default parameter to avoid **having to inject it**. In practice, we only create this object once, in ExpiryDateView and I don't envision another place to consume it.

Inside `ExpiryDateView` it gets created as:
```Swift
  init(cardValidator: CardValidator) {
    self.dateFormatter = ExpiryDateFormatter(cardValidator: cardValidator)
```

Reply:
I meant why have you hardcoded `.sandbox` inside CardValidator. Is that used for logging? if then what happens when our SDK is used in production?

Reply:
That is the default argument to the initialiser. If you create an object `ExpiryDateFormatter` and do not provide a parameter for `cardValidator`, it will fallback to this one.
In practice, we only use this object in 1 place (ExpiryDateView) and we do not rely on this default parameter, we inject a parameter.

The reason I provided the default parameter has been the unit testing, where having extensive tests that don't all require injecting a parameter, has allowed to save some space and hopefully keep those tests more concise.

So we do not "hardcode" the environment into the CardValidator. As to why that is injected in the first place, I have now looked into it and it does appear to be for logging. But this Validator has been inherited from 3.x versions and seems to interact with a singleton logger. That is however not the same logger as the one we use for frames, it's an obsolete one. We may add something to backlog to remove that logger altogether if that is not something we need, our codebase would be grateful for it.